react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class RegistrationForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: ''
+    };
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleUsernameChange(event) {
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  handlePasswordChange(event) {
+    this.setState({ [event.target.name]: event.target.value });
+  }
```

Comment:
That's a very general-purpose way of handling change events! However, in practice, part of a good user experience includes sanitizing field values on-the-fly (for example, preventing invalid characters from showing up or enforcing a maximum value length), which is often different per-field. Not important here, just something to be aware of with future applications.